### DISCUSSION TITLE
Now What?

### DISCUSSION THREAD

**Discussion by EvanColon** [sudoguy/tiktokpy#298] [Q&A]
I followed the directions on the readme to get it set up and working. What are some things I can do it after it says

| INFO | 📈 Getting trending items
📈 Getting trending en:   0%|                                                                  | 0/5 [07:03<?, ?it/s]

**Comment by TuanTool** [comment]
playwright._impl._errors.TimeoutError: Timeout 30000ms exceeded.
Can you explain my error ?

**Reply by uhaop** [reply]
i get the same error any way to fix? tried extending time out but doesn't seem o change anything

**Reply by uhaop** [reply]
FIgured it out replace the quickstart.py file with this code 


import asyncio
from tiktokpy import TikTokPy
from playwright.async_api import Error

async def main():
    async with TikTokPy() as bot:
        try:
            # Retry up to 3 times if a timeout occurs
            for _ in range(3):
                try:
                    # Do you want to get trending videos? You can!
                    trending_items = await asyncio.wait_for(bot.trending(amount=50), timeout=30)

                    for item in trending_items:
                        try:
                            # ❤️ you can like videos
                            await asyncio.wait_for(bot.like(item), timeout=30)
                            # or unlike them
                            await asyncio.wait_for(bot.unlike(item), timeout=30)
                            # or follow users
                            await asyncio.wait_for(bot.follow(item.author.username), timeout=30)
                            # as and unfollow
                            await asyncio.wait_for(bot.unfollow(item.author.username), timeout=30)
                        except asyncio.TimeoutError:
                            print("⚠️ Timeout occurred while trying to perform the operation")
                        except Error as e:
                            print(f"⚠️ An error occurred: {e}")

                    break  # Exit the retry loop if successful
                except asyncio.TimeoutError:
                    print("⚠️ Timeout occurred while getting trending items. Retrying...")
                except Error as e:
                    print(f"⚠️ An error occurred: {e}")

            # 😏 getting user's feed
            user_feed_items = await bot.user_feed(username="tiktok", amount=50)
            for item in user_feed_items:
                # 🎧 get music title, cover, link, author name..
                print("Music title: ", item.music.title)
                # #️⃣ print all tag's title of video
                print([tag.title for tag in item.challenges])
                # 📈 check all video stats
                print("Comments: ", item.stats.comments)
                print("Plays: ", item.stats.plays)
                print("Shares: ", item.stats.shares)
                print("Likes: ", item.stats.likes)

        except Error as e:
            print(f"⚠️ An error occurred: {e}")

asyncio.run(main())
